### PR TITLE
base/build: using `Libdl.dlext` for searching lib

### PR DESCRIPTION
**Pull request by iblis17**
Blocker: #271

### CODE REVIEW

**Comment by iblis17** [comment]
travis ci failure looks weird
```
g++: internal compiler error: Killed (program cc1plus)
Please submit a full bug report,
with preprocessed source if appropriate.
See <file:///usr/share/doc/gcc-4.8/README.Bugs> for instructions.
make: *** [build/src/operator/tensor/broadcast_reduce_op_index.o] Error 4
make: *** Waiting for unfinished jobs....
================================[ ERROR: MXNet ]================================
LoadError: failed process: Process(`make -j8 USE_BLAS=openblas 'MSHADOW_LDFLAGS=-lm /home/travis/julia/bin/../lib/julia/libopenblas64_.so'`, ProcessExited(2)) [2]
while loading /home/travis/.julia/v0.6/MXNet/deps/build.jl, in expression starting on line 58
================================================================================
```

File: deps/build.jl [hunk 1]
```diff
@@ -11,7 +11,8 @@ curr_win = "20170502"
 if haskey(ENV, "MXNET_HOME")
   info("MXNET_HOME environment detected: $(ENV["MXNET_HOME"])")
   info("Trying to load existing libmxnet...")
-  lib = Libdl.find_library(["libmxnet", "libmxnet.so"], ["$(ENV["MXNET_HOME"])/lib"])
+  lib = Libdl.find_library("libmxnet.$(Libdl.dlext)",
```
Comment (vchuravy) [inline deps/build.jl]: I believe under linux you need to continue to use `"libmxnet"` without the `Libdl.ext`

Comment (iblis17) [reply deps/build.jl]: seems both work.
I can add it back if you want it.

```
% uname -a
Linux kaladbolg 4.12.8-2-ARCH #1 SMP PREEMPT Fri Aug 18 14:08:02 UTC 2017 x86_64 GNU/Linux
% ls
libmxnet.so*  libmxnet-test.so*
```
```
julia> p = Libdl.find_library(["libmxnet-test.so"], [pwd()])        
"/home/iblis/.julia/v0.5/MXNet/deps/usr/lib/libmxnet-test.so"       

julia> Libdl.dlopen_e(p)          
Ptr{Void} @0x0000000002f4c7b0
```

Comment (vchuravy) [reply deps/build.jl]: Interesting. Then we need to figure out why Travis is failing.

**Comment by iblis17** [comment]
Travis kill the julia 0.6 build, due to no output in 10 mins: 
    https://travis-ci.org/dmlc/MXNet.jl/jobs/272416995

But I think the stdout of `make blablabla` (from `Pkg.build`) is gone in julia 0.6 and master.
please checkout
    * https://github.com/JuliaLang/julia/blame/f32499fa7ef8c29049a39344565011b10401b23b/base/pkg/entry.jl#L618
    * https://github.com/JuliaLang/julia/pull/20070#pullrequestreview-60873107

**Comment by iblis17** [comment]
I think travis's build hit another issue:
https://github.com/apache/incubator-mxnet/pull/7741#discussion_r137577886

**Comment by iblis17** [comment]
I will rebase this PR if this issue gets resolved: https://github.com/dmlc/MXNet.jl/pull/267

**Comment by vchuravy** [comment]
Can we split the travis related commits into their own pr?

**Comment by iblis17** [comment]
sure. It's #271

**Pull request merged**